bugfix/11993-wordcloud-setdata

kamilkulig opened the pull request:
Fixed #11993, wordcloud looked incorrect after running `setData()`.

circleci-mu commented:
## Visual test results - Differences found
Found **1** diffing sample(s). [Please review the differences.](https://vrevs.highsoft.com/pr/12891/review)

pawelfus commented on ts/modules/wordcloud.src.ts:
```diff
@@ -980,14 +980,21 @@ var wordCloudSeries: Partial<Highcharts.WordcloudSeries> = {
             }),
             field: Highcharts.WordcloudFieldObject;
 
+        // Reset the scale before finding the dimensions (#11993)
```
A comment why we need to reset the scale would be nice. It's quite unusual reason AFAIR ;)

kamilkulig commented:
The visual change is not deliberate. However, the reference image doesn't seem like the intended look in the first place. The position of tooltip isn't something that we can recreate in the demo just by moving the mouse:
 
![image](https://user-images.githubusercontent.com/13188589/73843933-e4523400-481f-11ea-8f80-7a1aab5f15a0.png)

Also, the words overlap in both images which leads me to suspicion that screenshot has been made during the initial animation of the series.

TorsteinHonsi commented:
> The visual change is not deliberate.

The test tool opens the first point's tooltip by `point.onMouseOver()`. Maybe the change in the sorting order explains why a different point is hovered now? Please check that, then come back with an advice on whether the visual change should be approved or not.

kamilkulig commented:
Now I see. I added `concat()` in my commit to preserve the original order in `series.points` array.  `series.setData()` doesn't work correctly without it.
After my change the tooltip displays for word `Lorem` - the first data point.  So, I corrected the demo and the visual change should be approved.

kamilkulig commented:
Now I see. I added `.concat()` in my commit to preserve the original order of `series.points` array. `.setData()` doesn't work correctly without it.
After my change, the tooltip displays for the word "Lorem" - the first point. So my commit corrects the demo and the visual change should be approved.
